### PR TITLE
Implement SMTP Configuration to Enhanced User Experience with Password Reset Email Notifications

### PR DESCRIPTION
**Pull request by shikharpa**
This pull request introduces SMTP configuration to enable email notifications for critical events within the backend authentication system. With this enhancement, the system gains the capability to dispatch emails promptly, ensuring users stay informed about important updates, such as password resets, account verifications.
This pull request also adds functionality to send email notifications for password reset requests. Users can now initiate a password reset process through the application, triggering an email containing a unique token for verification. Upon receiving the token, users can reset their passwords securely. This feature enhances user experience by providing a convenient and secure method for recovering forgotten passwords.
Screenshot 2024-02-23 111317

## Summary

- [x] Tested (Must)
- [ ] Test Case added
- [x] Build Successful (Must)
- [x] Sufficient Code comments added (Must)
- [x] Attached Screenshots / Videos 
- [ ] All Relevant Documents added

### CODE REVIEW

File: services/api/kalvi/settings.py [hunk 1]
```diff
@@ -159,3 +159,11 @@
 }
 
 PASSWORD_RESET_TIMEOUT=900     #900 Sec = 15 Min
+
+# Email Configuration
+EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+EMAIL_HOST = 'smtp.gmail.com'
```
Comment (yogai53) [inline services/api/kalvi/settings.py]: Grab the value from the environment variable. Through Docker compose. Refer how its done for DB setup

https://github.com/kalvilabs/kalvios/blob/master/services/api/kalvi/settings.py#L84

**Review by yogai53 — CHANGES_REQUESTED** [review]

File: services/api/kalvi/settings.py (hunk 1 above)
Comment (shikharpa) [reply services/api/kalvi/settings.py]: yes i had done that

File: services/api/db/serializers.py [hunk 2]
```diff
@@ -91,12 +94,14 @@ def validate(self, attrs):
         try:
             uid = urlsafe_base64_encode(force_bytes(user.id))
             token = PasswordResetTokenGenerator().make_token(user)
-        except Exception:
+            reset_link = reverse('reset-password-through-mail', kwargs={'uid': uid, 'token': token})
+        except ValidationError:
```
Comment (yaju09) [inline services/api/db/serializers.py]:
```suggestion
        except ValidationError:
```
will the exception type ValidationError be able to handle all possible exceptions for the code block involving uid, token and reset_link

File: services/api/db/templates/reset_password_email.html [hunk 3]
```diff
@@ -0,0 +1,1665 @@
...
+        line-height: 0;
+        max-height: 0;
+        mso-hide: all;
+        overflow: hidden;
+        table-layout: fixed;
+        visibility: hidden;
+        width: 0;
+      }
+    </style>
+    <!--[if mso
```
Comment (yaju09) [inline services/api/db/templates/reset_password_email.html]:
```suggestion
    <!--[if mso
```
please remove stale code

File: services/api/db/templates/reset_password_email.html [hunk 4]
```diff
@@ -0,0 +1,1665 @@
...
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="format-detection" content="telephone=no" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Set a new password to your Kalvios account</title>
+    <style type="text/css" emogrify="no">
+      #outlook a {
+        padding: 0;
+      }
+      .ExternalClass {
```
Comment (yaju09) [inline services/api/db/templates/reset_password_email.html]:
```suggestion
      .ExternalClass {
```
is the above class being used?

**Review by yaju09 — CHANGES_REQUESTED** [review]
is there any way to reduce the size of reset_password_email.html ?

File: services/api/db/utils.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+def send_reset_password_email(recipient_email, reset_url):
+    try:
+        connection = get_connection(
+        backend=settings.EMAIL_BACKEND,
+        host=settings.EMAIL_HOST,
+        port=settings.EMAIL_PORT,
+        username=settings.EMAIL_HOST_USER,
+        password=settings.EMAIL_HOST_PASSWORD,
+        use_tls=True, ) # or use_ssl=True for SSL
+        subject = "Reset Your Password on"
```
Comment (elizabeththomas92) [inline services/api/db/utils.py]: this subject text seems to be incomplete

**Review by elizabeththomas92 — CHANGES_REQUESTED** [review]

File: services/api/db/serializers.py (hunk 2 above)
Comment (shikharpa) [reply services/api/db/serializers.py]: Look at now I've added base Exception class to catch all possible issues.

File: services/api/db/templates/reset_password_email.html (hunk 3 above)
Comment (shikharpa) [reply services/api/db/templates/reset_password_email.html]: removed

File: services/api/db/templates/reset_password_email.html (hunk 4 above)
Comment (shikharpa) [reply services/api/db/templates/reset_password_email.html]: removed

File: services/api/db/utils.py (hunk 5 above)
Comment (shikharpa) [reply services/api/db/utils.py]: changed it

**Pull request closed without merging**